torch/csrc/jit/codegen/cuda/lower_expr_sort.cpp

```diff
@@ -0,0 +1,366 @@
+#include <torch/csrc/jit/codegen/cuda/lower_expr_sort.h>
+
+#include <torch/csrc/jit/codegen/cuda/ir_all_nodes.h>
+#include <torch/csrc/jit/codegen/cuda/ir_utils.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+namespace {
+
+TensorView* findOutputTensor(Expr* expr) {
+  TORCH_INTERNAL_ASSERT(
+      expr->outputs().size() <= 1, "Unexpected number of outputs");
+  if (expr->outputs().size() != 1) {
+    return nullptr;
+  }
+  auto out = expr->output(0);
+  if (out->getValType() != ValType::TensorView) {
+    return nullptr;
+  }
+  return out->as<TensorView>();
+}
+
+void findTargetTensor(Expr* expr, TensorView*& target, unsigned& score) {
+  TORCH_INTERNAL_ASSERT(expr->outputs().size() <= 1);
+
+  TensorView* out_tv = findOutputTensor(expr);
+  if (out_tv == nullptr) {
+    target = nullptr;
+    score = 0;
+    return;
+  }
+
+  if (!out_tv->hasComputeAt()) {
+    target = out_tv;
+    // No computeAt, so this should come last.
+    score = std::numeric_limits<unsigned>::max();
+    return;
+  }
+
+  // Note this returns the computeAt position
+  int pos = (int)out_tv->getRelativeComputeAtAxis();
+  target = out_tv->getComputeAtView();
+  while (target->hasComputeAt()) {
+    if ((int)target->getThisComputeAtAxis() < pos) {
+      break;
+    }
+    // getComputeAtRelPos accepts an axis index.
+    pos = pos == 0 ? 0 : target->getComputeAtRelPos(pos - 1) + 1;
+    target = target->getComputeAtView();
+  }
+
+  score = pos;
+}
+
+// Type definitions for brevity
+using ExprListT = std::vector<Expr*>;
```

Comment:
What does the T suffix mean?

Reply:
Nothing particular. Just my personal habit of appending `_t` in typedef'ed types in C.

Reply:
Would you mind cleaning the names? Since it's not part of the coding style, this local convention is more confusing than it helps.

Reply:
I removed the `T` suffix.